### PR TITLE
quick fix for delete.type, #4315 for next steps

### PR DESCRIPTION
**Pull request by sixfourtwelve**
## Overview

More of a temp solution to negate the issue. See https://github.com/unisonweb/unison/issues/4315 for the bigger/fuller fix

## Test coverage

Manual testing

### CODE REVIEW

File: unison-cli/src/Unison/CommandLine/InputPatterns.hs [hunk 1]
```diff
@@ -721,22 +721,22 @@ deleteGen suffix target mkTarget =
         )
 
 delete :: InputPattern
-delete = deleteGen Nothing "term or type" (DeleteTarget'TermOrType DeleteOutput'NoDiff)
+delete = deleteGen Nothing exactDefinitionTermQueryArg "term or type" (DeleteTarget'TermOrType DeleteOutput'NoDiff)
 
 deleteVerbose :: InputPattern
-deleteVerbose = deleteGen (Just "verbose") "term or type" (DeleteTarget'TermOrType DeleteOutput'Diff)
+deleteVerbose = deleteGen (Just "verbose") exactDefinitionTermQueryArg "term or type" (DeleteTarget'TermOrType DeleteOutput'Diff)
```
Comment (ChrisPenner) [inline unison-cli/src/Unison/CommandLine/InputPatterns.hs]:
```suggestion
deleteVerbose = deleteGen (Just "verbose") exactDefinitionArg "term or type" (DeleteTarget'TermOrType DeleteOutput'Diff)
```

**Review by ChrisPenner — APPROVED** [review]
Looks good to me 👍🏼

**Comment by ChrisPenner** [comment]
I think I maybe wasn't very clear about what I was asking for with the tab-completion tests, I think we've already got the actual behaviour of `delete.term` and `delete.type` covered elsewhere. The tests for autocompleting the commands them selves is good, but I meant that it'd be good to add some tests for tab-completing the *arguments* to these commands, something like this:

    ```unison
    structural type MyType = MyConstructor

    theTerm = 1
    ```

    ```ucm
    .> add
    .> debug.tab-complete delete.type My
    .> debug.tab-complete the
    ```

😄

File: unison-src/transcripts/tab-completion.md [hunk 2]
```diff
@@ -9,6 +9,22 @@ Test that tab completion works as expected.
...
+unique type Foo = A | B
+
+add : a -> a
+add b = b
+```
+
+```ucm
+.> update
+.> delete.type Foo
+.> delete.term add
```
Comment (aryairani) [inline unison-src/transcripts/tab-completion.md:27]: Were these meant to be `debug.tab-complete delete.type Foo` and `debug.tab-complete delete.type add`?

**Review by aryairani — COMMENTED** [review]
Thanks @sixfourtwelve, this looks nice! I left one question.

File: unison-src/transcripts/tab-completion.md (hunk 2 above)
Comment (aryairani) [reply unison-src/transcripts/tab-completion.md:27]: erp that's basically what Chris said also.

**Comment by sixfourtwelve** [comment]
https://github.com/unisonweb/unison/pull/4342
Closing for the above PR, apologies for the mess

**Pull request closed without merging**